### PR TITLE
Test k8s 1.25 1.26 on kind

### LINKED ISSUE #6571: Test 1.25 and 1.26 on kind
Kind v0.17.0 support both 1.25 and 1.26 with the following node images:

kindest/node:v1.25.3@sha256:f52781bc0d7a19fb6c405c2af83abfeb311f130707a0e219175677e366cc45d1
kindest/node:v1.26.0@sha256:691e24bd2417609db7e589e1a479b902d2e209892a10ce375fab60a8407c7352

Source: https://github.com/kubernetes-sigs/kind/releases

### PR DESCRIPTION
**Pull request by thbkrkr**
- Update kind to `v0.17.0` and node images for this version (see https://github.com/kubernetes-sigs/kind/releases)
- Add k8s `v1.25.3` and `v1.26.0`

Indirected related to this:
- Keep only GKE, updated to k8s `v1.25` in the deployer plans

Resolves #6571.

### CODE REVIEW

**Comment by thbkrkr** [comment]
buildkite test this -f p=kind,t=TestSmoke -m DEPLOYER_KIND_NODE_IMAGE=kindest/node:v1.25.3@sha256:f52781bc0d7a19fb6c405c2af83abfeb311f130707a0e219175677e366cc45d1,DEPLOYER_KIND_NODE_IMAGE=kindest/node:v1.26.0@sha256:691e24bd2417609db7e589e1a479b902d2e209892a10ce375fab60a8407c7352

File: .buildkite/e2e/nightly-main-matrix.yaml [hunk 1]
```diff
@@ -25,22 +25,16 @@
...
-    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.22.9@sha256:8135260b959dfe320206eb36b3aeda9cffcb262f4b44cda6b33f7bb73f453105
-      DEPLOYER_KIND_IP_FAMILY: ipv6
-    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.23.6@sha256:b1fa224cc6c7ff32455e0b1fd9cbfd3d3bc87ecaa8fcb06961ed1afb3db0f9ae
     - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.23.6@sha256:b1fa224cc6c7ff32455e0b1fd9cbfd3d3bc87ecaa8fcb06961ed1afb3db0f9ae
-      DEPLOYER_KIND_IP_FAMILY: ipv6
-    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.24.1@sha256:11276ce4763d5600379c0104c4cb51bf7420c8c02937708e958ad68d08e0910c
     - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.24.1@sha256:11276ce4763d5600379c0104c4cb51bf7420c8c02937708e958ad68d08e0910c
+    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.25.3@sha256:f52781bc0d7a19fb6c405c2af83abfeb311f130707a0e219175677e366cc45d1
+    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.26.0@sha256:691e24bd2417609db7e589e1a479b902d2e209892a10ce375fab60a8407c7352
+    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.26.0@sha256:691e24bd2417609db7e589e1a479b902d2e209892a10ce375fab60a8407c7352
```
Comment (barkbay) [inline .buildkite/e2e/nightly-main-matrix.yaml:32]: nit: duplicated?
```suggestion
```

File: .buildkite/e2e/nightly-main-matrix.yaml [hunk 2]
```diff
@@ -25,22 +25,16 @@
...
     - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.22.9@sha256:8135260b959dfe320206eb36b3aeda9cffcb262f4b44cda6b33f7bb73f453105
-    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.22.9@sha256:8135260b959dfe320206eb36b3aeda9cffcb262f4b44cda6b33f7bb73f453105
-      DEPLOYER_KIND_IP_FAMILY: ipv6
-    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.23.6@sha256:b1fa224cc6c7ff32455e0b1fd9cbfd3d3bc87ecaa8fcb06961ed1afb3db0f9ae
     - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.23.6@sha256:b1fa224cc6c7ff32455e0b1fd9cbfd3d3bc87ecaa8fcb06961ed1afb3db0f9ae
-      DEPLOYER_KIND_IP_FAMILY: ipv6
-    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.24.1@sha256:11276ce4763d5600379c0104c4cb51bf7420c8c02937708e958ad68d08e0910c
     - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.24.1@sha256:11276ce4763d5600379c0104c4cb51bf7420c8c02937708e958ad68d08e0910c
+    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.25.3@sha256:f52781bc0d7a19fb6c405c2af83abfeb311f130707a0e219175677e366cc45d1
+    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.26.0@sha256:691e24bd2417609db7e589e1a479b902d2e209892a10ce375fab60a8407c7352
```
Comment (barkbay) [inline .buildkite/e2e/nightly-main-matrix.yaml:31]: nit: [1.26.2 is available](https://hub.docker.com/layers/kindest/node/v1.26.2/images/sha256-c39462fc9f460e13627cbd835b7d1268e4fd1a82d23833864e33ac1aaa79ee7a?context=explore)

File: .buildkite/e2e/nightly-main-matrix.yaml (hunk 1 above)
Comment (thbkrkr) [reply .buildkite/e2e/nightly-main-matrix.yaml:32]: This looks like a duplicate but the second has `DEPLOYER_KIND_IP_FAMILY: ipv6` in addition. 
```
    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.25.3@sha256:..
    
    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.26.0@sha256:..

    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.26.0@sha256:..
      DEPLOYER_KIND_IP_FAMILY: ipv6
```

**Comment by thbkrkr** [comment]
buildkite test this -f p=kind,t=TestUpdate,DEPLOYER_KIND_NODE_IMAGE=kindest/node:v1.26.2@sha256: c39462fc9f460e13627cbd835b7d1268e4fd1a82d23833864e33ac1aaa79ee7a -m DEPLOYER_CLIENT_VERSION=0.15.0,DEPLOYER_CLIENT_VERSION=0.17.0

**Comment by thbkrkr** [comment]
buildkite test this -f p=kind,t=TestUpdate,DEPLOYER_KIND_NODE_IMAGE=kindest/node:v1.26.2@sha256:c39462fc9f460e13627cbd835b7d1268e4fd1a82d23833864e33ac1aaa79ee7a -m DEPLOYER_CLIENT_VERSION=0.15.0,DEPLOYER_CLIENT_VERSION=0.17.0

File: .buildkite/e2e/nightly-main-matrix.yaml (hunk 2 above)
Comment (thbkrkr) [reply .buildkite/e2e/nightly-main-matrix.yaml:31]: I will test it. Not sure it is compatible with Kind 0.15.0, which by the way could be updated.

**Comment by thbkrkr** [comment]
@elasticmachine run elasticsearch-ci/docs

**Comment by thbkrkr** [comment]
buildkite test this -f p=kind -m DEPLOYER_KIND_NODE_IMAGE=kindest/node:v1.19.16@sha256:476cb3269232888437b61deca013832fee41f9f074f9bed79f57e4280f7c48b7,DEPLOYER_KIND_NODE_IMAGE=kindest/node:v1.20.15@sha256:a32bf55309294120616886b5338f95dd98a2f7231519c7dedcec32ba29699394,DEPLOYER_KIND_NODE_IMAGE=kindest/node:v1.21.14@sha256:9d9eb5fb26b4fbc0c6d95fa8c790414f9750dd583f5d7cee45d92e8c26670aa1,DEPLOYER_KIND_NODE_IMAGE=kindest/node:v1.22.15@sha256:7d9708c4b0873f0fe2e171e2b1b7f45ae89482617778c1c875f1053d4cef2e41,DEPLOYER_KIND_NODE_IMAGE=kindest/node:v1.23.13@sha256:ef453bb7c79f0e3caba88d2067d4196f427794086a7d0df8df4f019d5e336b61,DEPLOYER_KIND_NODE_IMAGE=kindest/node:v1.24.7@sha256:577c630ce8e509131eab1aea12c022190978dd2f745aac5eb1fe65c0807eb315,DEPLOYER_KIND_NODE_IMAGE=kindest/node:v1.25.3@sha256:f52781bc0d7a19fb6c405c2af83abfeb311f130707a0e219175677e366cc45d1

**Comment by thbkrkr** [comment]
buildkite test this -f p=kind,t=TestSmoke -m DEPLOYER_KIND_NODE_IMAGE=kindest/node:v1.19.16@sha256:476cb3269232888437b61deca013832fee41f9f074f9bed79f57e4280f7c48b7,DEPLOYER_KIND_NODE_IMAGE=kindest/node:v1.20.15@sha256:a32bf55309294120616886b5338f95dd98a2f7231519c7dedcec32ba29699394,DEPLOYER_KIND_NODE_IMAGE=kindest/node:v1.21.14@sha256:9d9eb5fb26b4fbc0c6d95fa8c790414f9750dd583f5d7cee45d92e8c26670aa1,DEPLOYER_KIND_NODE_IMAGE=kindest/node:v1.22.15@sha256:7d9708c4b0873f0fe2e171e2b1b7f45ae89482617778c1c875f1053d4cef2e41,DEPLOYER_KIND_NODE_IMAGE=kindest/node:v1.23.13@sha256:ef453bb7c79f0e3caba88d2067d4196f427794086a7d0df8df4f019d5e336b61,DEPLOYER_KIND_NODE_IMAGE=kindest/node:v1.24.7@sha256:577c630ce8e509131eab1aea12c022190978dd2f745aac5eb1fe65c0807eb315,DEPLOYER_KIND_NODE_IMAGE=kindest/node:v1.25.3@sha256:f52781bc0d7a19fb6c405c2af83abfeb311f130707a0e219175677e366cc45d1

**Pull request merged**